doc/source/reference/functions/string_slice.rst

```diff
@@ -0,0 +1,199 @@
+.. -*- rst -*-
+
+.. highlightlang:: none
+
+.. groonga-command
+.. database: functions_string_slice
+
+``string_slice``
+==================
+
+.. versionadded:: 11.0.2
+
+Summary
+-------
+
+``string_slice`` は文字列の部分文字列を抽出します。引数によって部分文字列の抽出方法を使い分けることが出来ます。抽出方法は以下の2つがあります。
+
+* 位置による抽出
+* 正規表現による抽出
+
+Groongaでは正規表現にRubyと同じ構文を使います。
+
+この関数で正規表現による抽出をする場合、正規表現のマッチにはインデックスを使用しません。
+また、検索で正規表現を使う場合と異なり、マッチ対象のテキストを正規化しません。 :doc:`/reference/regular_expression` も参照してください。
+
+To enable this function, register ``functions/string`` plugin by following the command::
+
+  plugin_register functions/string
+
+Syntax
+------
+
+``string_slice`` は2つから4つの引数を指定できます。抽出方法によって指定可能な引数が変わります。
+
+位置による抽出
+^^^^^^^^^^^^^^^^^^
+
+::
+
+  string_slice(target, nth[, options])``
+  string_slice(target, nth, length[, options])``
+
+``options`` には以下のキーを指定します。すべてのキー・値のペアは省略可能です。::
+
+  {
+    "default_value": default_value,
+  }
+
+正規表現による抽出
+^^^^^^^^^^^^^^^^^^^^^^
+ 
+::
+
+  string_slice(target, regexp, nth_or_name[, options])
+
+
+``options`` には以下のキーを指定します。すべてのキー・値のペアは省略可能です。::
+
+  {
+    "default_value": default_value,
+  }
+
+
+Usage
+-----
+
+Here are a schema definition and sample data to show usage.
+
+Sample schema:
+
+.. groonga-command
+.. include:: ../../example/reference/functions/string_slice/usage_setup_schema.log
```

Comment:
You don't need to generate this file manually. You can just put commands here (see `doc/source/reference/commands/select.rst` for example):

```suggestion
.. include:: ../../example/reference/functions/string_slice/usage_setup_schema.log
.. plugin_register functions/string
.. table_create Memos TABLE_HASH_KEY ShortText
```
and generate execution example by the following command line:

```bash
cd doc
make update-examples SOURCES=source/reference/functions/string_slice.rst
```

But this will not work on Windows... We need to improve our CMake configuration to support documentation.

We should also improve our developer documentation...

Reply:
Yes, please.